src/Analysis/Ast/Impl/Dependencies/DependencyGraph.cs

```diff
@@ -131,12 +131,13 @@ internal sealed class DependencyGraph<TKey, TValue> {
             var missingKeysHashSet = new HashSet<TKey>();
             for (var i = 0; i < _verticesByIndex.Count; i++) {
                 var vertex = _verticesByIndex[i];
-                var newIncoming = ImmutableArray<int>.Empty;
+                var newIncomingArr = vertex.IncomingKeys;
```

Comment:
My only worry here is that this is a copy and has a performance impact; there's no way for me to refer to this array except to do a full name everywhere (which I can do), but when I profiled it, it didn't seem to matter one way or the other, so I left this for readability reasons. Happy to change if it's bad.

Reply:
this also looks like you can use ImmutableArray.Builder.

also, you can actually pool Builder to even remove some more cost?